thingworx-import/Entities.xml

```diff
@@ -426,130 +465,243 @@ function run() {
-    command = 'sendoffline?name=' + me.name;
+    // Check if property exists/needs to be created
+    var needToCreateProperty = false;
+    try {
+        var propertyLogging = Things[thingName].GetPropertyLogging({
+            propertyName: tagName
+        });
+    } catch (err) {
+        needToCreateProperty = true;
+    }
 
-  params = {
-    url: encodeURI(host + command + credentials) /* STRING */,
-  };
+    // Create property if necessary
+    if (needToCreateProperty) {
+        // Get property data change type for creation
+        var propDataChangeType = "VALUE";
+        if (me.useDataChangeTypeAlways) {
+            propDataChangeType = "ALWAYS";
+        }
+        
+        Things[thingName].AddPropertyDefinition({
+            readOnly: false,
+            remote: false,
+            type: tagType,
+            logged: true,
+            name: tagName,
+            persistent: true,
+            dataChangeType: propDataChangeType
+        });
+        Things[thingName].RestartThing();
+    }
 
-  // sends the api request to DataMailBox
-  result = Resources['ContentLoaderFunctions'].GetJSON(params);
-  logger.warn(JSON.stringify(result));
-  if (!result.success) {
+    // Convert tag value to proper type
+    var tagValFinal = tagValue;
+    if (tagType === "INTEGER") {
+        tagValFinal = parseInt(tagValue);
+    }
+
+    // Update property value
+    var createITParams = {
+        infoTableName: "InfoTable",
+        dataShapeName: "NamedVTQ"
+    };
+    var properties = Resources["InfoTableFunctions"].CreateInfoTableFromDataShape(createITParams);
+    var row = new Object();
+    row.name = propName;
+    row.value = tagValFinal;
+    row.quality = tagQuality;
+    row.time = new Date(timestamp);
+    properties.AddRow(row);
+    Things[thingName].UpdatePropertyValues({
+        values: properties
+    });
+
+
+
+}
+if (tagName && tagValue && timestamp) {
+    run();
+} else {
     logger.error(
-      'Error calling DataMailbox API in SendEwonOffline. Error: ' +
-        result['message']
+        'One or more parameters for InsertDataPoint were undefined or null.'
+    );
+    logger.debug(
+        'tagName: ' +
+        tagName +
+        'tagValue: ' +
+        tagValue +
+        'timestamp: ' +
+        timestamp
+    );
+}]]></code></Row></Rows></ConfigurationTable></ConfigurationTables></ServiceImplementation><ServiceImplementation description="" handlerName="Script" name="ProcessTimeSinceUpdate"><ConfigurationTables><ConfigurationTable dataShapeName="" description="" isMultiRow="false" name="Script" ordinal="0"><DataShape><FieldDefinitions><FieldDefinition baseType="STRING" description="code" name="code" ordinal="0"/></FieldDefinitions></DataShape><Rows><Row><code><![CDATA[/*
+ * Keeps track of the time since the last data update was received. If the time
+ * is long enough, then an error is sent.
+ *
+ * @author HMS Networks, MU Americas Solution Center
+ */
+const MILLISECONDS_IN_MINUTES = 60000;
```

Comment:
This should probably be MILLISECONDS_PER_MINUTE

Reply:
Same comment: "That can be changed, although the code in this service was copy/pasted from the ConnectorHost version of the service, which is nearly identical. The original should probably be updated as well."

Reply:
Changed